source/postcard-schema/tests/max_size.rs

```diff
@@ -0,0 +1,28 @@
+#![allow(dead_code)]
+
+use postcard_schema::{max_size::max_size, Schema};
+
+#[derive(Schema)]
+enum ExampleE {
+    Foo,
+    Bar([u8; 5]),
+    Baz {
+        a: Result<i32, u32>,
+        b: i128,
+    }
+}
+
+#[derive(Schema)]
+struct ExampleS {
+    a: Result<i32, u32>,
+    b: i128,
+    c: bool,
+}
+
+#[test]
+fn smoke() {
+    let max1 = max_size::<ExampleE>();
+    let max2 = max_size::<ExampleS>();
+    assert_eq!(Some(34), max1);
```

Comment:
This actually overestimates Result a bit - we should count the number of variants and use that to figure out how many varint bits it would be, for example if we only have 2 variants, we only need to add 1 byte for the discriminant, not the full 5.